parameters.h

```diff
@@ -27,29 +27,42 @@
 #include <stdint.h>
 #include <string>
 #include <vector>
+#include "arch/arch_device_api.h"
 #include <map>
 
 #include "definitions.h"
 
+
 const uint64_t INVALID_CELLID = 0;
 
+struct GridParameters {
```

Comment:
Here also renaming to FSgrid params, and retaining the xmin etc in the below Parameters struct.